### ISSUE TITLE
Automatic API data sanitization

### ISSUE THREAD

**Issue by nikosdion** [akeeba/panopticon#341]
The most common failure mode we see with the Joomla! JSON API calls is that a third party plugin emits junk before the start of the JSON response – either directly, or by triggering PHP messages (warnings, notices, strict notices, deprecated notices, or user notices). This typically happens because either the third party plugin or Joomla! itself (groan!) is not fully compatible with newer PHP 8 versions. Since Joomla! loads these behaviour, system, and activity log plugins before our web services plugin there's nothing we can do to avoid the output of junk, especially when error reporting is set to Development in Global Configuration.

In those cases, we observe that the junk is followed by _valid JSON data_ we can consume. If only there was a way…

Well, this is exactly what we've been doing in the Akeeba Backup JSON API connector library code. We expect that there will be junk, and we remove it. 

The exact code we use there cannot be used directly in Panopticon when talking to the Joomla! API as that code is based on the assumption of simple, flat objects. The Joomla! API returns nested objects which break these assumptions.

However, the concept is solid. We can use `json_validate` (or a polyfill, for PHP 8.1 and 8.2) to go through chunks of the returned raw data, narrowing it down by opening braces (expected return type: JSON object) or opening brackets (expected return type: JSON array). This is not very expensive, and can salvage connections to site which are otherwise hosed by PHP messages preceding the usable JSON data.

In short, we can salvage the connection with sites suffering the most common failure mode (a plugin is soiling the output) by scrubbing the junk which precedes the usable API data.
